plugins/catalog-backend/src/catalog/types.ts

```diff
@@ -158,3 +168,108 @@ export interface EntitiesCatalog {
    */
   facets(request: EntityFacetsRequest): Promise<EntityFacetsResponse>;
 }
+
+/**
+ * The request shape for {@link EntitiesCatalog.paginatedEntities}.
+ *
+ * @alpha
+ */
+export type PaginatedEntitiesRequest =
+  | PaginatedEntitiesInitialRequest
+  | PaginatedEntitiesCursorRequest;
+
+/**
+ * The initial request for {@link EntitiesCatalog.paginatedEntities}.
+ * The request take immutable properties that are going to be bound
+ * for the current and the next pagination requests.
+ *
+ * @alpha
+ */
+export interface PaginatedEntitiesInitialRequest {
+  authorizationToken?: string;
+  fields?: (entity: Entity) => Entity;
+  limit?: number;
+  filter?: EntityFilter;
+  sortField?: string;
+  query?: string;
+  sortFieldOrder?: 'asc' | 'desc' | undefined;
+}
+
+/**
+ * Request for {@link EntitiesCatalog.paginatedEntities} used to
+ * move forward or backward on the data.
+ *
+ * @alpha
+ */
+export interface PaginatedEntitiesCursorRequest {
+  authorizationToken?: string;
+  fields?: (entity: Entity) => Entity;
+  limit?: number;
+  cursor: string;
+}
+
+/**
+ * The response shape for {@link EntitiesCatalog.paginatedEntities}.
+ *
+ * @alpha
+ */
+export interface PaginatedEntitiesResponse {
+  /**
+   * The entities for the current pagination request
+   */
+  entities: Entity[];
+  /**
+   * The cursor of the next pagination request.
+   */
+  nextCursor?: string;
+  /**
+   * The cursor of the previous pagination request.
+   */
+  prevCursor?: string;
+  /**
+   * the total number of entities matching the current filters.
+   */
+  totalItems: number;
+}
+
+/**
+ * The Cursor used internally by the catalog.
+ *
+ * @alpha
+ */
+export type Cursor = {
```

Comment:
Since we are using a cursor based pagination, changing the filters while using a cursor doesn’t make any sense, since changing filters affect the matched entities and pagination itself.
So, we will end up in a situation where the cursor (which was previously sent with the previous api call and different filters) produces weird behaviors making the pagination not linear.
This is why I’ve kept only on the initial request. The same applies to the sort fields and sort field order.

If we think changing filters "on the go" is important, we should pivot back to a limit/offset pagination.

Reply:
yeah thinking some more about it it probably makes sense, and if we want to support switching out the filters while supplying a cursor we'd leave that to be figured out by the backend rather than frontend

Reply:
Agreed, we could even (in the future, not necessary right now perhaps) reject attempts at changing filter/sorting params during cursor pagination

Reply:
> Agreed, we could even (in the future, not necessary right now perhaps) reject attempts at changing filter/sorting params during cursor pagination

note that the only way to reject these attempts is to store the initial filters in the cursor

Reply:
Yeah totally, I was probably not clear - I agreed with the design of putting these parameters in the cursor :)